macro-pdfviewer-api/src/main/java/com/xwiki/pdfviewer/internal/macro/PDFViewerMacro.java

```diff
@@ -48,29 +65,56 @@
 @Singleton
 public class PDFViewerMacro extends AbstractMacro<PDFViewerMacroParameters>
 {
+    private final List<String> delegatedRightsValues = List.of("1", "true", "yes");
+
+    @Inject
+    private AuthorizationManager authorizationManager;
+
     @Inject
     private TemplateManager templateManager;
 
+    @Inject
+    @Named("resource/standardURL")
+    private EntityReferenceResolver<String> urlEntityReferenceResolver;
+
+    @Inject
+    @Named("current")
+    private EntityReferenceResolver<String> entityReferenceResolver;
+
     @Inject
     private ScriptContextManager scriptContextManager;
 
+    @Inject
+    private Provider<XWikiContext> wikiContextProvider;
+
+    @Inject
+    private Provider<MJSFilter> mjsFilterProvider;
+
     /**
      * Create and initialize the descriptor of the macro.
      */
     public PDFViewerMacro()
     {
         super("PDF Viewer", "View PDF attachments inside wiki pages without downloading or importing them.",
             PDFViewerMacroParameters.class);
+        // Error when building the project on 14.10.
+//        setDefaultCategory(DEFAULT_CATEGORY_CONTENT);
     }
 
     @Override
     public List<Block> execute(PDFViewerMacroParameters parameters, String content, MacroTransformationContext context)
         throws MacroExecutionException
     {
+        mjsFilterProvider.get().maybeAddMJSMimeType();
```

Comment:
I'd catch the exception here and provide the steps to register the mime type manually.